src/sentry/integrations/github/repository.py

```diff
@@ -65,25 +36,25 @@ def validate_config(self, organization, config):
         return config
         ```
         """
-        if config.get('name') and config.get('installation'):
+        if config.get('identifier') and config.get('installation'):
             integration = Integration.objects.get(
                 id=config['installation'], organizations=organization)
             installation = integration.get_installation(organization.id)
             client = installation.get_client()
 
-            repo = self._validate_repo(client, installation, config['name'])
+            repo = self._validate_repo(client, installation, config['identifier'])
             config['external_id'] = six.text_type(repo['id'])
```

Comment:
Should the `external_id` be the `repo['id]` for github and not the `full_name`? just checking I don't actually know. :)

Reply:
Eh I guess it could be. I'm not sure if we just decided to keep the id incase we needed it for some reason? since we already have the name anyway. @macqueen thoughts?

Reply:
repo name can change in github, so it should be the id

Reply:
Sigh this repo/full_name rather than just using id can be rather confusing......... I assume the repo_id then is only unique for the user specified. Ok best to leave it as is.